hal/src/nRF52840/deviceid_hal.cpp

```diff
@@ -78,6 +79,19 @@ int hal_get_device_serial_number(char* str, size_t size, void* reserved)
     return HAL_DEVICE_SERIAL_NUMBER_SIZE;
 }
 
+int hal_get_device_hw_version()
+{
+    // HW Data format: | NCP_ID | HW_VERSION | HW Feature Flags |
+    //                 | byte 0 |   byte 1   |    byte 2/3      |
+    uint8_t hw_data[4] = {};
+    int r = hal_exflash_read_special(HAL_EXFLASH_SPECIAL_SECTOR_OTP, HW_DATA_OTP_ADDRESS, hw_data, 4);
+
+    if (r != 0 || hw_data[1] == 0xFF) {
+        return -1;
```

Comment:
`system_error_t` codes should be used instead.

Reply:
Fixed